### PR TITLE
Add make init

### PR DESCRIPTION
**Pull request by costinm**

### CODE REVIEW

File: prow/e2e-suite.sh [hunk 1]
```diff
@@ -62,6 +62,7 @@ ISTIO_GO=$(cd $(dirname $0)/..; pwd)
 export HUB=${HUB:-"gcr.io/istio-testing"}
 export TAG="${GIT_SHA}"
 
+make init
```
Comment (ldemailly) [inline prow/e2e-suite.sh]: let's add init to where it's missing in the makefile 

```
ldemailly-macbookpro:istio ldemailly$ \rm -rf vendor/
ldemailly-macbookpro:istio ldemailly$ time make init
~/go/out ~/go/src/istio.io/istio
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 74.0M  100 74.0M    0     0  23.8M      0  0:00:03  0:00:03 --:--:-- 23.8M
~/go/src/istio.io/istio

real	0m21.812s
user	0m13.226s
sys	0m28.736s
ldemailly-macbookpro:istio ldemailly$ time make init
make: Nothing to be done for `init'.

real	0m0.096s
user	0m0.040s
sys	0m0.041s
```

Comment (costinm) [reply prow/e2e-suite.sh]: If this fixes the prow problem - we can do what you propose in a separate PR and evaluate what impact it has on the build time.

**Comment by costinm** [comment]
BTW - with the build optimizations (removing the redundant builds) circle has finished the build, lint, unit and e2e_simple test before prow got even the presubmit.

**Comment by ldemailly** [comment]
if even half of the effort spent on circle is spent on prow, similar results could be achieved, it's not like circle is magic, it's just a cute UI with a very convenient "ssh" feature - and ideally the makefile gains would be for both systems not exclusively tweaked for circle

**Comment by ozevren** [comment]
@ldemailly, @costinm 

Guys, are we decided on this? I want to make sure we fix the issue in a timely manner.

**Comment by costinm** [comment]
If the test passes - I think we should submit it.

If Laurent wants to implement a different solution, and if it doesn't degrade performance - he can do it in a separate PR.

**Comment by ozevren** [comment]
Prow folks, anyone would like to chime-in and lgtm?

**Pull request merged**